tasks.py

```diff
@@ -0,0 +1,200 @@
+        c.run(
+            f"docker login ghcr.io -u {config.github_user} --password-stdin",
+            in_stream=pwd,
+            echo=True,
+        )
+        c.run(f"docker pull {config.image_path}:{config.docker_version}", echo=True)
+        logging.info("Bring down docker containers.")
+        c.run(
+            f"DOCKER_IMAGE_TAG={config.docker_version} docker compose -f docker-compose.prod-img.yml down",
+            echo=True,
+        )
+        logging.info("Get code changes from repo")
+        pull_repository(c, config)
+        logging.info("Run necessary migrations.")
+        migrate(c, config)
+        logging.info("Bring docker service back up.")
+        c.run(
+            f"DOCKER_IMAGE_TAG={config.docker_version} docker compose -f docker-compose.prod-img.yml up -d",
+            echo=True,
+        )
+        logging.info("Success!")
+
+
+@task(
+    help={
+        "environment": "'staging' or 'prod' indicating which environment to target",
+        "github_user": "github user name to authenticate with",
+        "github_token": "token to use together with github_user",
+    }
+)
+def deploy(c, environment, github_user, github_token):
+    """Deploy the most recent docker image for the given environment."""
+    config = get_configs(environment, github_user, github_token)
+    connection = config.get_connection()
+    deploy_(connection, config)
+
+
+def backup_(c: Connection, config: HostConfig):
+    """Create a local encrypted backup of the sql database."""
+    logging.info("Start creating database backup.")
+    with c.cd(config.code_dir):
+        c.run(
+            f"""echo 'bash -c "pg_dump $SQLALCHEMY_DATABASE_URI -f /backup/backup.sql"' | docker run --env-file=.env --rm -i -v {config.backup_dir}:/backup/ postgres bash"""
+        )
+    logging.info("gzip backup.")
+    backup_path = os.path.join(config.backup_dir, "backup.sql")
+    c.run(f"gzip -f {backup_path}")
+    backup_timestamped_name = os.path.join(
+        config.backup_dir,
+        f"{config.backup_name}_{datetime.datetime.now().strftime('%Y%m%d-%H%M%S')}.gz.gpg",
+    )
+    logging.info("Encrypt backup.")
+    c.run(
+        f"gpg --output {backup_timestamped_name} --encrypt --trust-model always --batch --no-tty --recipient {config.key_id} {backup_path}.gz"
+    )
+    c.run(f"rm {backup_path}.gz")
+    logging.info("Success!")
+
+
+@task(
+    help={"environment": "'staging' or 'prod' indicating which environment to target"}
+)
+def backup(c, environment):
+    """Create a local encrypted backup of the sql database."""
+    config = get_configs(environment)
+    connection = config.get_connection()
+    backup_(connection, config)
+
+
+def cleanup_(c: Connection, config: HostConfig):
+    """Execute cleanup tasks, usually after deployment"""
+    logging.info("Remove unused docker images")
+    c.run("docker image prune -f")
+
+
+@task(
+    help={"environment": "'staging' or 'prod' indicating which environment to target"}
```

Comment:
Nit: It looks like we use this string at least 3 times, so it'd be worth saving the key and value to a constant, imo.

Reply:
I was thinking about that. There is the downside of the help strings potentially not being updated if something changes, but the plus is that it is also a comment to anyone reading the code, so I am leaning to keep it as it is

Reply:
I'm here for that.